pkg/util/dynamichelper/gvrresolver_test.go

```diff
@@ -195,17 +199,35 @@ func TestFindGVR(t *testing.T) {
 		},
 	}
 
-	for _, tt := range tests {
+	for index, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			r := &gvrResolver{apiresources: tt.resources}
 
 			got, err := r.Resolve(tt.kind, "")
-			if !reflect.DeepEqual(err, tt.wantErr) {
-				t.Errorf("got: %#v, expected: %#v", err, tt.wantErr)
+			if err != nil {
+				for _, i := range deep.Equal(err.Error(), tt.wantErr.Error()) {
+					t.Error(i)
+				}
```

Comment:
As mentioned in https://github.com/Azure/ARO-RP/pull/2806#discussion_r1184194712 `deep.Equal` on the error type is an anti-pattern. This will probably need to be changed before the PR gets approved, however it's not clear to me what the replacement should be. If you know, please comment here!

Reply:
not sure if I am understanding that in the right way, maybe we should just directly compare the error strings?

```go
if err != nil && testCase.wantErr != nil {
			if !strings.EqualFold(err.Error(), testCase.wantErr.Error()) {
				t.Errorf(...)
			}
			return
}
```

Reply:
The problem is that we shouldn't be doing comparisons on the error message itself from my understanding. We either should be using custom error types or checking only a portion of the error strings which is interesting in-and-of-itself. At least that's what I got from the aforementioned comment's link.

Reply:
@ArrisLee is identifying what we've traditionally done in this codebase, string comparison. Per the original linting PR I linked, that's only somewhat better than deep equal, a final, more mature error state would be error enriching, which is way out of scope.

Granted, let's keep this out of the PR for now, I mostly meant it as an observation. We should create a new work item to track that fix.

Reply:
you should probably use the function in `test/util/error/error.go` for error checking. In any case here I am not sure what `deep.Equal` value added is, since `Error()` returns a string.